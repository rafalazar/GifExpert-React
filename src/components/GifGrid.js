import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import GifGridItem from './GifGridItem';
import PropTypes from 'prop-types';

const GifGrid = ({category}) => {

    const {data:images,loading} = useFetchGifs(category);
    console.log(loading);

    return(
        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>

            {loading ? <p className="animate__animated animate__flash">Loading...</p> : null}

            <div className="card-grid">
                
                {
                    images.map(img => {
                        return <GifGridItem 
                            key={img.id}
                            {...img}
                        />
                    })
                }
                
            </div>
        </>
    );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}

export default GifGrid;