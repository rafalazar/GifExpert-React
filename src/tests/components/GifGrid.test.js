import React from "react";
import { shallow } from "enzyme";
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe("Pruebas sobre <GifGrid />", () => {

    const category = "The last of us";

    test("Debe mostrar el componente correctamente", () => {

        useFetchGifs.mockReturnValue({

            data: [],
            loading: true,

		});

		const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();

    });

    test("Debe mostrar items cuando se cargan imÃ¡genes con useFetchGifs", () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Whatever'
        }];

        useFetchGifs.mockReturnValue({
			data: gifs,
			loading: false
		});

		const wrapper = shallow(<GifGrid category={category} />);
		
		expect(wrapper.find('p').exists()).toBe(false);

    });

});
