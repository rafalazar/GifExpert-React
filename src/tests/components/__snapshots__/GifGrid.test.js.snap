// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <GifGrid /> Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    The last of us
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
